index.js

```diff
@@ -117,15 +118,21 @@ module.exports = class Transaction {
    * @return {Buffer}
    */
   hash (signature) {
+    const rawCopy = this.raw.slice(0)
+    if (this._chainId) {
+      this.v = this._chainId
+      this.r = this.s = 0
+    }
     let toHash
 
-    if (typeof signature === 'undefined') {
+    if (this._chainId || typeof signature === 'undefined') {
       signature = true
     }
 
     toHash = signature ? this.raw : this.raw.slice(0, 6)
 
     // create hash
+    this.raw = rawCopy.slice(0)
```

Comment:
the reason why I added that is to make hashing function not destructive, for example if you initialize the Tx object with preexisting v,r,s values calling hash will set r,s values to 0 and v will be set based on the chain and initial v,r,s values will be lost. If we can ignore this condition let me know :)